### ISSUE TITLE
Lots of images: make sprite every x img

### ISSUE THREAD

**Issue by kud** [Compass/compass#876]
Hey,

Imagine I've 400 images in folder/ and I can't insert all these images in one sprite (too big). I would like to create a new sprite every 20 images.

I've done compass sprite "folder/*.png" to see if I can custom anything but I don't see where.

The thing is I would like to keep the idea to have just .icon-img-{id} in my css because I find it great but I need to call a different sprite when I'm in .icon-img-21.

Do you have any idea?

Thanks.

Regards,

_kud

**Comment by chriseppstein** [comment]
I would like to support this use case by allowing you to a) enumerate what sprite files should be in a particular sprite map, and b) allow you iterate over the sprites in a folder.

Then you can support this case as well as several others. 

Hunt & pecked on my iPhone... Sorry if it's brief!

**Comment by kud** [comment]
The thing is I mostly would like not to see any different in css, because the current listing is perfect. It's just that every 20 time (a variable so), you create a xxx-yy.zzz.png sprite where xxx is the folder name, yy the "pagination", and zzz the generated id.

You of course have to reset the background-position every time you change the sprite. Hmmm.

**Comment by chriseppstein** [comment]
Right. I understand that would make this one use case the most optimal, but I'd rather build the underlying support necessary to accomplish this first, then if this pattern becomes common we can optimize it by adding a core mixin.

**Comment by cw-moshe** [comment]
Your concern is actually about the size and not the number. A more generic solution will be to rotate every xKB. And I would vote for this as well.

**Comment by kud** [comment]
It's true. +1

**Comment by ventrebleu** [comment]
+1

**Comment by ifthenelse** [comment]
+1
